superset/utils/decorators.py

```diff
@@ -116,3 +118,20 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def logged_in_api(f: Callable) -> Callable:
+    """
+        Use this decorator to require a user to be logged to use your API methods.
+
+        If the user is not logged in, this will return an HTTP 401.
+    """
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if not g.user.get_id():
```

Comment:
Out of interest when is the user not logged in? Is this not a guaranteed thing?

Reply:
if the cookie expires or you just curl the url then they wouldn't be logged in